runtime/compiler/trj9/Makefile

```diff
@@ -37,6 +37,19 @@ ifeq ($(SPEC),linux_arm)
     $(warning ARM SPEC detected)
 endif
 
+# During openJDK build, SPEC arrives as 
+# /root/openj9-openjdk-jdk9/build/linux-arm-normal-server-release/spec.gmk
+# and PLATFORM is not set, so the above doesn't work.
+ifdef OPENJ9_CC_PREFIX
```

Comment:
I don't think we should be depending on something defined by the environment, better to (indirectly) depend on something declared by the buildspec which can be converted into an export from mkconstants.mk.ftl

Reply:
or actually runtime/compiler/makefile.ftl

Reply:
Hi @pshipton , it's an interesting point. The details of the cross compiler tool chain are known in (and indeed set by) the Dockerfile, so it seems natural to push that information from the Dockerfile into the build. Any mechanism that doesn't take the information from the Dockerfile is clearly both more work and an opportunity to introduce an error. Using environment variables defined by the Dockerfile is simply the most obvious way of passing the information from Docker to the build, but I guess we could do it any number of other ways. Do you object specifically to using env vars or more generally to passing the information from Docker?

Reply:
Perhaps I'm too stuck on old ways of doing things. In the future there will be a configure step that takes settings from the environment or by parameter, even though up to now its been controlled by the build specs. Although to do this correctly it seems flags like `-march=armv6 -marm -mfpu=vfp -mfloat-abi=hard` must also be derived from the OPENJ9_CC_PREFIX. 

How does the OpenJDK compile determine which versions of the tools to use?

Reply:
> it seems flags like -march=armv6 -marm -mfpu=vfp -mfloat-abi=hard must also be derived from the OPENJ9_CC_PREFIX


Not quite, a given toolchain (as defined by prefix) can build for many arch targets, and indeed I have another PR in the works for switching between armv6 and armv7. In the future we may also need to support v5 for embedded applications, aarch64 for servers and softfloat for some subset of the above.
OpenJDK hasn't (yet?) embraced Docker files, so the tool chain has to be specified via options to their configure script, e.g.
```
bash ./configure --openjdk-target=arm-linux-gnueabihf --with-abi-profile=arm-vfp-hflt --with-x=../../gcc-linaro-4.9.4-2017.01-x86_64_arm-linux-gnueabihf/arm-linux-gnueabihf/ --with-freetype=/home/jkingdon/tr/openJDK/gcc-linaro-4.9.4-2017.01-x86_64_arm-linux-gnueabihf/arm-linux-gnueabihf/libc/usr/
```
I opened https://github.com/eclipse/openj9/issues/518 to address better integration with the open jdk configure options, and would much prefer to address that work there rather than blocking a working ARM build.

I created a new build spec as part of this set of PRs (this is only one of many required to get the build working on ARM), and it seems to me that adding new specs is a considerable amount of work with the potential for introducing errors. I'd be hesitant about any solution that is based on a ballooning number of build-specs.

One possibility for passing configuration information between Docker and the build is to add scripts to the image that contain the relevant info. For example, that long winded configure commandline could be wrapped in a script and packed in the Docker image. This would make more sense after #518 has been addressed.

Reply:
> and would much prefer to address that work there rather than blocking a working ARM build

+1